chain/types/actor_event.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// what value codec does client want to match on ?
+	Codec uint64 `json:"codec"`
+	// data associated with the "event key"
+	Value []byte `json:"value"`
+}
+
+type SubActorEventFilter struct {
+	Filter  ActorEventFilter `json:"filter"`
+	Prefill bool             `json:"prefill"`
+}
+
+type ActorEventFilter struct {
+	// Matches events from one of these actors, or any actor if empty.
+	// TODO: Should we also allow Eth addresses here?
+	// For now, this MUST be a Filecoin address.
+	Addresses []address.Address `json:"address"`
```

Comment:
Should we allow clients to specify both ETH & Filecoin addresses to filter by in this new API ?

Reply:
Does this work now if you just bundle up an eth address into an `address.Address` here, or do we have something that stops that being useful later on? Since an `ethtypes.EthAddress` can also be an `address.Address`, can we just handle it all in one? Or do we have precedent for handling eth addresses differently on external APIs?

Reply:
I would only support Filecoin-formatted addresses here for now. Also parsing for EthAddresses here and converting could be nice, but:
- I suspect we'll run into unexpected edge-cases
- It strikes me as strictly an enhancement, so let's descope from this.

Reply:
Yeh let's just go ahead with Filecoin addresses for now and revisit later if support ETH addresses becomes necessary.